### PR TITLE
Module source pinned ref check

### PR DESCRIPTION
**Pull request by OmarDarwish**
This is to add a warning to modules from source control that either do not pin a version or pin to a default version. Documentation is provided [here](https://github.com/OmarDarwish/tflint/blob/7734d73d1abff1e110bf3befe86277e6c73f982d/docs/module_pinned_source.md)

### CODE REVIEW

**Comment by LudwigHoff** [comment]
+1

File: detector/detector.go [hunk 1]
```diff
@@ -168,7 +167,24 @@ func (d *Detector) Detect() []*issue.Issue {
...
+	}
+
+	for name, m := range modules {
+		if d.Config.IgnoreModule[m.Source] {
+			d.Logger.Info(fmt.Sprintf("ignore module `%s`", name))
+			continue
+		}
+		// Special case for module_pinned_source rule
+		modulePinnedSourceRule := "module_pinned_source"
+		if d.Config.IgnoreModule[modulePinnedSourceRule] {
```
Comment (wata727) [inline detector/detector.go]: Is this `d.Config.IgnoreRule` ?

File: detector/detector.go [hunk 2]
```diff
@@ -168,7 +167,24 @@ func (d *Detector) Detect() []*issue.Issue {
...
+		}
+	}
+
+	for name, m := range modules {
+		if d.Config.IgnoreModule[m.Source] {
+			d.Logger.Info(fmt.Sprintf("ignore module `%s`", name))
+			continue
+		}
+		// Special case for module_pinned_source rule
+		modulePinnedSourceRule := "module_pinned_source"
```
Comment (wata727) [inline detector/detector.go]: Please rename to `terraform_module_pinned_source` :)

Comment (OmarDarwish) [inline detector/detector.go]: Done!

File: detector/detector.go (hunk 1 above)
Comment (OmarDarwish) [inline detector/detector.go]: Yep. Changed.

File: detector/detector.go [hunk 3]
```diff
@@ -168,7 +167,24 @@ func (d *Detector) Detect() []*issue.Issue {
...
+			continue
+		}
+		// Special case for terraform_module_pinned_source rule
+		modulePinnedSourceRule := "terraform_module_pinned_source"
+		if d.Config.IgnoreRule[modulePinnedSourceRule] {
+			d.Logger.Info(fmt.Sprintf("ignore module `%s`", modulePinnedSourceRule))
+			continue
 		}
+		d.Logger.Info(fmt.Sprintf("run module linter `%s`", name))
+		modulePinnedSourceDetector := NewTerraformModulePinnedSourceDetector(d, m.File, m.ObjectItem)
```
Comment (wata727) [inline detector/detector.go]: __[gometalinter]__
* [interfacer] undeclared name: NewTerraformModulePinnedSourceDetector
* [structcheck] unused struct field undeclared name: NewTerraformModulePinnedSourceDetector
* [unconvert] undeclared name: NewTerraformModulePinnedSourceDetector
* [staticcheck] undeclared name: NewTerraformModulePinnedSourceDetector
* [gosimple] undeclared name: NewTerraformModulePinnedSourceDetector
* [varcheck] unused variable or constant undeclared name: NewTerraformModulePinnedSourceDetector

Comment (OmarDarwish) [inline detector/detector.go]: Fixed

File: docs/README.md [hunk 4]
```diff
@@ -28,6 +28,8 @@ Issues are classified into the following three types.
 - **AWS Route**
     - [aws_route_not_specified_target](aws_route_not_specified_target.md)
     - [aws_route_specified_multiple_targets](aws_route_specified_multiple_targets.md)
+- **Terraform**
+    - [module_pinned_source](module_pinned_source.md)
```
Comment (wata727) [inline docs/README.md]: Please also rename this link and title.

**Comment by wata727** [comment]
@OmarDarwish Great work! I'm very pleased to add such a useful detector!

**Comment by LudwigHoff** [comment]
+1

**Pull request merged**